database_cleanup/remove_old_order_references/remove_old_order_references.sh

```diff
@@ -0,0 +1,34 @@
+mongo "${MONGO_URL}/production" --eval '
+db.userdetails.aggregate([
+    {
+        $lookup: {
+            from: "orders",
+            localField: "orders",
+            foreignField: "_id",
+            as: "matched_orders"
+        }
+    },
+    {
+        $addFields: {
+            orders: {
+                $filter: {
+                    input: "$orders",
+                    as: "order",
+                    cond: { $in: ["$$order", "$matched_orders._id"] }
+                }
+            }
+        }
+    },
+    {
+       $project: {
+           matched_orders: 0,
+       }
+    },
+    {
+        $merge: {
+            into: "userdetails",
+            whenMatched: "replace"
+        }
+    }
```

Comment:
This is a possibly less destructive way to do it; it will only change the `orders` field. You can decide if you prefer it or not. I do think the `whenNotMatched: "discard"` should be included either way, as we otherwise may restore deleted objects. Just for good practice.

```suggestion
    {
        $project: {
            orders: 1,
        },
    },
    {
        $merge: {
            into: "userdetails",
            whenNotMatched: "discard"
        },
    }
```